From 5addb3e1d2e8660cc771b5d4a5623a5cd279f3d0 Mon Sep 17 00:00:00 2001
From: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
Date: Fri, 27 Oct 2023 10:45:53 +0530
Subject: [PATCH] mtd: rawnand: qcom: Cleanup qpic driver for serial nand

This change will cleanup qpic driver for serial nand.

-> Add clear_bam_transaction() for NAND_CMD_READID_SERIAL
since we are calling multiple times read_id() from serial
nand device configuration path, we have to clear bam
transaction for read_id() operations , else it will mislead
some other bam related issue.

(cherry picked from commit 2d89d9bbd11b31744d76a8c9a998f6075dd76622)
Signed-off-by: Md Sadre Alam <quic_mdalam@quicinc.com>

Change-Id: I2929515bda3ab942a3d6beff305dd5ace372b9b5
Signed-off-by: Poovendhan Selvaraj <quic_poovendh@quicinc.com>
---
 drivers/mtd/nand/raw/qcom_nandc.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/drivers/mtd/nand/raw/qcom_nandc.c b/drivers/mtd/nand/raw/qcom_nandc.c
index 2eb0200236b0..52cc44bcfe23 100644
--- a/drivers/mtd/nand/raw/qcom_nandc.c
+++ b/drivers/mtd/nand/raw/qcom_nandc.c
@@ -1733,7 +1733,8 @@ static void pre_command(struct qcom_nand_host *host, int command)
 	clear_read_regs(nandc);
 
 	if (command == NAND_CMD_RESET || command == NAND_CMD_READID ||
-	    command == NAND_CMD_PARAM || command == NAND_CMD_ERASE1)
+	    command == NAND_CMD_PARAM || command == NAND_CMD_ERASE1 ||
+		command == NAND_CMD_READID_SERIAL)
 		clear_bam_transaction(nandc);
 }
 
-- 
2.34.1

